Configure.cmake

```diff
@@ -42,6 +45,28 @@ CHECK_C_SOURCE_COMPILES("
     __m128d r = _mm_mul_pd(_mm_set1_pd(1), _mm_set1_pd(2)); }"
   COMPILER_SUPPORTS_SSE2)
 
+set (CMAKE_REQUIRED_FLAGS ${FLAGS_ENABLE_SSE4})
+CHECK_C_SOURCE_COMPILES("
+  #if defined(_MSC_VER)
+  #include <intrin.h>
+  #else
+  #include <x86intrin.h>
+  #endif
+  int main() {
+    __m128d r = _mm_floor_sd(_mm_set1_pd(1), _mm_set1_pd(2)); }"
+  COMPILER_SUPPORTS_SSE4)
+
+set (CMAKE_REQUIRED_FLAGS ${FLAGS_ENABLE_AVX})
```

Comment:
Ok then, I am rolling back to do a try_compile and will add a little comment to document the Cygwin issue.